model/PermissionsService.php

```diff
@@ -58,28 +57,49 @@ public function __construct(
     public function saveResourcePermissionsRecursive(
         core_kernel_classes_Resource $resource,
         array $privilegesToSet
-    ):void {
+    ): void {
         $this->saveResourcePermissions($resource, $privilegesToSet, true);
     }
 
     private function saveResourcePermissions(
         core_kernel_classes_Resource $resource,
-        array $privilegesToSet, bool $isRecursive
+        array $privilegesToSet,
+        bool $isRecursive
     ): void {
-        $currentPrivileges = $this->dataBaseAccess->getResourcePermissions($resource->getUri());
-        $addRemove = $this->strategy->normalizeRequest($currentPrivileges, $privilegesToSet);
+        $resourceURI = $resource->getUri();
+
+        $this->debug(
+            'saveResourcePermissions resource=%s recursive=%s: privileges=%s',
+            $resourceURI,
+            $isRecursive ? 'true' : 'false',
+            var_export($privilegesToSet, true)
+        );
+
+        $currentPrivileges = $this->dataBaseAccess->getResourcePermissions($resourceURI);
+        $addRemove = $this->strategy->normalizeRequest(
+            $currentPrivileges,
+            $privilegesToSet
+        );
+
+        $this->debug('addRemove: %s', var_export($addRemove, true));
```

Comment:
- [ ]  Do not forget to remove or add better log message

Reply:
Changed: 70680c72d4c08d00213f10b678e4e42ba5219129